python/test_infra/ttir_builder.py

```diff
@@ -267,24 +267,26 @@ def eltwise_proxy(
             len(stack) > 0
         ), "Top of callstack to builder funcs must be outside this file"
 
+        id = self.get_next_global_id()
+
+        # FIXME: this should be a `Location.file`, but for some reason it causes
+        # strange decomposition inheritance behaviour that breaks using this as
+        # a key into the golden map
+        loc = Location.name(
+            stack[0].filename + ":" + str(stack[0].lineno) + ":id(" + str(id) + ")"
+        )
```

Comment:
Can we roll this entire stack trace + loc generation into a separate helper function?  We will soon want to add support for matmul and only non-`eltwise_proxy` ops and they will need this same loc info.

Reply:
Sure, it will be a bit different since the whole thing is based on the call stack itself, but I am pretty sure I can make it work